content/backend/typescript-apollo/9-deployment.md

```diff
@@ -0,0 +1,692 @@
+
+### Create new scripts in `package.json` 
+
+So far, you have been using `ts-node` to run your TypeScript code. However, this has a memory and performance overhead. In production, you will transpile your code to JavaScript using the TypeScript compiler and run the JavaScript code directly using `node`. To do this, you will need to create two additional scripts in your `package.json`. You will also need to add a new script to apply migrations to your production database. 
+
+
+<Instruction> 
+
+Update your `package.json` with the following scripts:
+
+```json{5-7}(path=".../hackernews-typescript/package.json") 
+  "scripts": {
+    "dev": "ts-node-dev --transpile-only --no-notify --exit-child src/index.ts",
+    "generate": "ts-node --transpile-only src/schema",
+    "prettier-format": "prettier 'src/**/*.ts' --write",
+    "migrate:deploy": "prisma migrate deploy",
+    "build": "prisma generate && npm run generate && tsc",
+    "start": "node dist/src/index.js"
+  },
+```
+
+</Instruction>
+
+Here's what the scripts are doing:
+
+1. The `migrate:deploy` script is used to apply all pending migrations to the production database by running `prisma migrate deploy`. This command is somewhat different from `prisma migrate dev`, which you have been using to generate/apply migrations in your development database. You can find more information about the `migrate deploy` command in the [Prisma docs](https://www.prisma.io/docs/concepts/components/prisma-migrate#production-and-testing-environments).
+2. The `build` script is used to build the app from scratch. This involves three steps: first running `prisma generate` to create the Prisma Client, then running `src/schema.ts` to generate the nexus assets, and finally running the TypeScript compiler (or `tsc`) to transpile your TypeScript code to JavaScript. 
+3. The `start` command runs the JavaScript code transpiled by `tsc` using `node`. Notice that the code generated by `tsc` goes into the `dist` folder because of the `outDir` option specified in your `tsconfig.json`. 
+
+Now you will create another commit. 
+
+<Instruction>
+
+Add your project source files to git and create a new commit:
+
+```bash(path=".../hackernews-typescript/")
+git add .
+git commit -m "update index.ts and create new scripts in package.json"
+```
+
+</Instruction>
+
+With that out of the way, it's time to get started deploying the app! 🚀
+
+
+### Creating resources on Heroku 
+
+In this section, you will create the Heroku app along with a PostgreSQL database to host your API. Make sure you have created a Heroku account and logged into the CLI with the `heroku login` command before proceeding with this section. 
+
+<Instruction>
+
+Inside the root folder of your project, start a new Heroku app with the following command. _Before running the command, make sure to replace "app-name-placeholder" with a meaningful name for your app._
```

Comment:
I think "app-name-placeholder" makes it very explicit that this is a placeholder name that needs to be replaced. In addition, the GitHub repo and directory are already named "hackernews-typescript". So I wanted the app name to be totally different from the repository name, to avoid any confusion. 

I think this is fine as it is.